File: src/modifiers/monospacedDigit.ts
import { createModifier } from './createModifier';

/// SwiftUI `.monospacedDigit()`. Keeps the default font but renders digits
/// with uniform (monospaced) width, so numbers don't shift horizontally as
/// they change — ideal for counters and timers. Takes no parameters.
export function monospacedDigit() {
  return createModifier('monospacedDigit');
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/MonospacedDigit.swift
import SwiftUI

/// SwiftUI `.monospacedDigit()`. Keeps the default font but renders digits
/// with uniform (monospaced) width so numbers don't shift horizontally as
/// they change. Takes no parameters.
enum RNWMonospacedDigitModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("monospacedDigit") { view, _, _ in
            AnyView(view.monospacedDigit())
        }
    }
}
